File: mail/put_msg_via_smtp.pl
#!/usr/bin/perl -w

=k

using: ./putmsg.pl eml/message/3/bad_links.eml

deps:
1) sudo yum -y install perl-Authen-SASL
2) perl-SMTP
3) perl-SMTP-SSL

=cut

use Net::SMTP;
use Net::SMTP::SSL;
use strict;

if (not defined $ARGV[0])
{
    print "need filename as cmd argument\n";
    exit 0;
}

my $u = 'email@domain.com';
my $p = 'pass';
my $smptp = 'smtp.mail.ru';

my ($host,$username,$password) = ($smptp, $u, $p);

my @data;
open my $fh,"<$ARGV[0]" or die "cant open";

while (<$fh>) {
    s/^To: [^\n\r]*/To: $username/;
    s/^ReSent-From: [^\n\r]*?(,?\r?)$/ReSent-From: $username$1/;
    push @data,$_;
}
close $fh;
shift @data if $data[0] =~ /From .*\n/;

my $smtp = ($host ne '173.194.71.109') ? Net::SMTP->new($host,Debug=>1) : Net::SMTP::SSL->new($host, Port => 465,Debug=>1);
unless ($smtp) {
    print "error $@\n";
    exit 0;
}
$smtp->auth($username,$password) or print "!!! NO AUTHORIZATION !!!\n";
$smtp->mail($username);
$smtp->to($username);
#$smtp->to('k.valiev@corp.mail.ru');
$smtp->data();
foreach (@data) {
    $smtp->datasend($_);
}
$smtp->datasend();
$smtp->quit;

File: mail/put_msg_via_smtp.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# example: ./send_emails_to_csv_list_via_smtp.py /home/o2gy/test.csv ejudge.bmstu@mail.ru password

import sys
import csv
from email.mime.text import MIMEText
from email.parser import Parser
import smtplib
import time

try:
    f = sys.argv[1]
except:
    print "enter msg to send"
    sys.exit()

try:
    user = sys.argv[2]
    password = sys.argv[3]
except:
    print "need user and pass to sign up on smtp: ./send_emails_to_csv_list_via_smtp.py msg.eml vasya@mail.ru password"
    sys.exit()

smtp_host   = 'smtp.mail.ru'
smtp_port   = 465

def send_email(message, login, passwd):

    #headers = Parser().parsestr(message)
    #print 'To: %s' % headers['to']
    #print 'From: %s' % headers['from']
    #msg = MIMEText(message.encode('utf-8'), 'html', 'utf-8')
    #msg = MIMEText(message)
    #msg["From"] = login
    #msg["To"] = login
    #msg["Subject"] = headers['subject']

    try:
        session = smtplib.SMTP_SSL(smtp_host, smtp_port)
        session.set_debuglevel(1)
        session.ehlo()
        session.login(login, passwd)
        session.sendmail(login, login, message)
        session.quit()
        print "Success"
    except smtplib.SMTPException:
        print "Error: unable to send"

with open(f, 'rb') as file:
    msg = file.read()
    file.close()
    send_email(msg, user, password)
